ec/provider_config_test.go

```diff
@@ -103,6 +102,19 @@ func Test_verboseSettings(t *testing.T) {
 }
 
 func Test_newAPIConfig(t *testing.T) {
+	// This is necessary to avoid any EC_API_KEY which might be set to cause
+	// test flakyness.
+	if k := os.Getenv("EC_API_KEY"); k != "" {
```

Comment:
Is this common? I've not really come across this issue?

Maybe we can extract it into it's own function? I think it'll make it look a bit cleaner, what do you think?

Reply:
Yeah it's pretty common if you're running the acceptance tests locally and have an `EC_API_KEY` set, then the tests will fail. Maybe you're not running the acceptance tests very often these days 🙊, haha just kidding!

Reply:
I'll extract it into its own function with a `t.Helper() set`

Reply:
acceptance tests 😖